extra/innochecksum.cc

```diff
@@ -1861,8 +1872,14 @@ int main(
 				printf("page " UINT32PF " ", cur_page_num);
 			}
 
-			if (page_get_page_no(buf) % physical_page_size == 0) {
+			/* Since we read the page0 already. We can
+			safely ignore if the page number is 0 as
+			extent descriptor page */
+			if (uint32_t page_no = page_get_page_no(buf)) {
+			    if (page_no > 0 &&
+			        page_no % physical_page_size == 0) {
 				memcpy(xdes, buf, physical_page_size);
+			    }
```

Comment:
The condition `page_no > 0 &&` is redundant, because the outer `if` already guarantees that (it is nonzero and an unsigned type).

We seem to be trusting the page number that is written to the page frame, instead of consulting `cur_page_num`. If we are not validating `page_get_page_no()` against `cur_page_num` anywhere, it would be good to add another command line option that checks that the page number and tablespace ID within the file are sensible, except when we get a page that is full of NUL bytes.